src/nxt_cgroup.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) Andrew Clayton
+ * Copyright (C) F5, Inc.
+ */
+
+#include <nxt_main.h>
+
+#include <nxt_cgroup.h>
+
+
+static int nxt_mk_cgpath_relative(nxt_task_t *task, const char *dir,
+    char *cgpath);
+static nxt_int_t nxt_mk_cgpath(nxt_task_t *task, const char *dir,
+    char *cgpath);
+
+
+nxt_int_t
+nxt_cgroup_proc_add(nxt_task_t *task, nxt_process_t *process)
+{
+    int        len;
+    char       cgprocs[NXT_MAX_PATH_LEN];
+    FILE       *fp;
+    nxt_int_t  ret;
+
+    if (task->thread->runtime->type != NXT_PROCESS_MAIN
+        || nxt_process_type(process) != NXT_PROCESS_PROTOTYPE
+        || process->isolation.cgroup.path == NULL)
+    {
+        return NXT_OK;
+    }
+
+    ret = nxt_mk_cgpath(task, process->isolation.cgroup.path, cgprocs);
+    if (nxt_slow_path(ret == NXT_ERROR)) {
+        return NXT_ERROR;
+    }
+
+    ret = nxt_fs_mkdir_all((const u_char *) cgprocs, 0777);
+    if (nxt_slow_path(ret == NXT_ERROR)) {
+        return NXT_ERROR;
+    }
+
+    len = strlen(cgprocs);
+
+    len = snprintf(cgprocs + len, NXT_MAX_PATH_LEN - len, "/cgroup.procs");
+    if (nxt_slow_path(len >= NXT_MAX_PATH_LEN)) {
+        nxt_errno = ENAMETOOLONG;
+        return NXT_ERROR;
+    }
```

Comment:
Uhhh.  This seems to be asking for `stpecpy()` (as defined here: <https://software.codidact.com/posts/285946/287522#answer-287522>).  But yeah, we don't have that API.

Reply:
But we don't, so...